scripts/check_vet.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+set -e
+
+for dir in $(scripts/moduledirs.sh); do
+    (
+      cd $dir
+      # Before go1.13, there was a bug that prevented the toolchain from
+      # finding a module when no buildable files were present.
+      # In go1.14 this is detected and the message states that build
+      # constraints exclude all Go files.
+      go list -json 2>&1 | grep -E 'build constraints exclude|cannot find module' > /dev/null
+      [ "$?" = "0" ] && echo "skipping $dir: no buildable Go files" && exit 0
```

Comment:
@axw Unfortunately, the go vet documentation states:

>Vet's exit code is non-zero for erroneous invocation of the tool

which I interpret as "if you're running it in a dir with no go files, that's erroneous".

This is checking for the list of files from `go list`, and checking to see if it meets the "no buildable files" condition; if it does we just continue along.

Note: Technically, you can have buildable test files and no buildable source files, but that doesn't seem like a sane thing to ever happen.

Reply:
the same failures start showing up in `make test`, sticking with adding a `doc.go` file for now